src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        let newUrl = this.win.location.href;
+        if (incomingFragment == "") {
+          if (newUrl.indexOf('#') == -1) {
+            newUrl = newUrl + '#.' + outgoingFragment;
+          } else {
+            newUrl = newUrl.replace('#', '#.' + outgoingFragment + '&');
+          }
+        } else {
+          newUrl = newUrl.replace(incomingFragment, outgoingFragment);
+        }
+        this.win.history.replaceState({}, '', newUrl);
```

Comment:
No, this is the actual API `Location.replace(string)` - it should do what you'd like it to do. You don't need to assign location afterwards. It'd just look like this:

```
this.win.location.replace(newUrl);
```